### PR TITLE
feat: current member

### PR DESCRIPTION
**Pull request by rebelchris**
Added a function to retrieve the current logged in users role for the requested squad.

### CODE REVIEW

File: src/schema/sources.ts [hunk 1]
```diff
@@ -273,6 +279,16 @@ export const typeDefs = /* GraphQL */ `
       """
       token: String
     ): Source! @auth
+
+    """
+    Removes the logged-in user as a member from the source
+    """
+    leaveSource(
```
Comment (idoshamun) [inline src/schema/sources.ts:286]: Did you branch out from the leave source branch, perhaps?

File: src/schema/sources.ts [hunk 2]
```diff
@@ -117,6 +118,11 @@ export const typeDefs = /* GraphQL */ `
     Number of members in the source
     """
     membersCount: Int!
+
+    """
+    Logged Member role
+    """
+    currentMember: String
```
Comment (idoshamun) [inline src/schema/sources.ts:125]: I think it would be more consistent to return a `SourceMember` object here. What do you think?

Comment (rebelchris) [reply src/schema/sources.ts:125]: Hmm we could, but only really care about the role for that specific squad.
What other props would you be interested in?

Comment (idoshamun) [reply src/schema/sources.ts:125]: Right now it's only this, but for future compatibility it's better to use the object.
We might want to present join date to the squad or notifications settings in the future and so on. Using the source member as an object here will give you support in all of them out of the box, whereas using only role may become legacy.

**Comment by rebelchris** [comment]
@idoshamun Added a relationalfield instead

File: __tests__/sources.ts [hunk 3]
```diff
@@ -186,6 +186,37 @@ query SourceByFeed($data: String!) {
   });
 });
 
+describe('query source current member', () => {
```
Comment (idoshamun) [inline __tests__/sources.ts:189]: Can you please add a test of an anonymous user and user which is not a source member?

File: src/schema/sources.ts [hunk 4]
```diff
@@ -118,6 +119,11 @@ export const typeDefs = /* GraphQL */ `
     Number of members in the source
     """
     membersCount: Int!
+
+    """
+    Logged-in member object
+    """
+    currentMember: SourceMember!
```
Comment (idoshamun) [inline src/schema/sources.ts:126]: It can be null

```suggestion
    currentMember: SourceMember
```

File: __tests__/sources.ts (hunk 3 above)
Comment (rebelchris) [reply __tests__/sources.ts:189]: It should fail right? as it's trying to query role which is required on a member?

Comment (idoshamun) [reply __tests__/sources.ts:189]: I'd say return null. We wouldn't want the whole query to fail in this case.
It's a valid use case for public squads for example

Comment (rebelchris) [reply __tests__/sources.ts:189]: Yep agreed, modified the query and test cases now.

**Review by idoshamun — APPROVED** [review]
Looks great!

**Pull request merged**